open_event/api/sessions.py

```diff
@@ -137,19 +137,31 @@ def fix_payload_post(self, event_id, data):
     def update(self, event_id, service_id, data):
         data = self.validate(data, event_id)
         data_copy = data.copy()
+        st = data_copy.get('session_type')
         data_copy = self.fix_payload_post(event_id, data_copy)
         data = self._delete_fields(data)
         obj = ServiceDAO.update(self, event_id, service_id, data, validate=False)
         obj.track = data_copy['track']
         obj.microlocation = data_copy['microlocation']
         obj.speakers = data_copy['speakers']
-        obj.session_type = data_copy['session_type']
+        obj.session_type = st
         obj = save_db_model(obj, SessionModel.__name__, event_id)
         return obj
 
     def create(self, event_id, data, url):
         data = self.validate(data, event_id)
         payload = self.fix_payload_post(event_id, data)
+        # Handle SessionTypes separately
+        st_data = data.get('session_type')
+        if st_data:
+            session_type = SessionTypeModel(
+                name=st_data['name'],
+                length=st_data['length'],
+                event_id=event_id
+            )
+            save_to_db(session_type, 'SessionType saved')
+            payload['session_type'] = session_type
+
         return ServiceDAO.create(self, event_id, payload, url, validate=False)
```

Comment:
If you do it this way, then a session type will be created for each session. So if we have just 2 session types in the whole event and there are 100s of sessions, then 100s of duplicate session types will be created. This is not we want.

Also what about `session_type_id` in POST model. I see it is now not being used.
I suggest you leave the sessions API as it was and rather make change in the import_helpers to import session types from sessions json.